### PR TITLE
Fix dotnetcore tests and cleanup

### PR DESCRIPTION
**Pull request by boblangley**

### CODE REVIEW

File: src/NServiceBus.Autofac.AcceptanceTests/NServiceBus.Autofac.AcceptanceTests.csproj [hunk 1]
```diff
@@ -1,18 +1,21 @@
 ﻿<?xml version="1.0" encoding="utf-8"?>
 <Project Sdk="Microsoft.NET.Sdk">
   <PropertyGroup>
-    <TargetFrameworks>net452;netcoreapp2.0</TargetFrameworks>
-    <DebugSymbols>True</DebugSymbols>
-    <NoWarn>PCR0001</NoWarn>
+    <TargetFrameworks>net462;netcoreapp2.0</TargetFrameworks>
+    <TreatWarningsAsErrors>true</TreatWarningsAsErrors>
+    <CopyLocalLockFileAssemblies>true</CopyLocalLockFileAssemblies>
```
Comment (SimonCropp) [inline src/NServiceBus.Autofac.AcceptanceTests/NServiceBus.Autofac.AcceptanceTests.csproj]: out of curiosity why `CopyLocalLockFileAssemblies`?

File: src/NServiceBus.Autofac.AcceptanceTests/NServiceBus.Autofac.AcceptanceTests.csproj [hunk 2]
```diff
@@ -1,18 +1,21 @@
 ﻿<?xml version="1.0" encoding="utf-8"?>
 <Project Sdk="Microsoft.NET.Sdk">
   <PropertyGroup>
-    <TargetFrameworks>net452;netcoreapp2.0</TargetFrameworks>
-    <DebugSymbols>True</DebugSymbols>
-    <NoWarn>PCR0001</NoWarn>
+    <TargetFrameworks>net462;netcoreapp2.0</TargetFrameworks>
+    <TreatWarningsAsErrors>true</TreatWarningsAsErrors>
```
Comment (SimonCropp) [inline src/NServiceBus.Autofac.AcceptanceTests/NServiceBus.Autofac.AcceptanceTests.csproj]: isnt `TreatWarningsAsErrors` already configured?
 https://github.com/Particular/NServiceBus.Autofac/blob/develop/src/Directory.Build.props#L4

File: src/NServiceBus.Autofac/NServiceBus.Autofac.csproj [hunk 3]
```diff
@@ -3,27 +3,26 @@
...
   <ItemGroup>
-    <PackageReference Include="Autofac" Version="[4.0.0, 5.0.0)" />
-    <PackageReference Include="Janitor.Fody" Version="1.*" PrivateAssets="All" />
-    <PackageReference Include="NServiceBus" Version="[7.0.0-*, 8.0.0)" />
-    <PackageReference Include="Particular.CodeRules" Version="0.2.0" PrivateAssets="All" />
+    <PackageReference Include="Autofac" Version="[4.0.0,5.0.0)" />
+    <PackageReference Include="NServiceBus" Version="[7.0.0-*,8.0.0)" />
+    <PackageReference Include="Particular.CodeRules" Version="0.2.0" PrivateAssets="all" />
     <PackageReference Include="Fody" Version="2.1.2" PrivateAssets="All" />
+    <PackageReference Include="Janitor.Fody" Version="1.5.1" PrivateAssets="All" />
```
Comment (SimonCropp) [inline src/NServiceBus.Autofac/NServiceBus.Autofac.csproj]: why not a wildcard?

File: src/NServiceBus.Autofac.AcceptanceTests/NServiceBus.Autofac.AcceptanceTests.csproj (hunk 1 above)
Comment (WilliamBZA) [reply src/NServiceBus.Autofac.AcceptanceTests/NServiceBus.Autofac.AcceptanceTests.csproj]: > Add <CopyLocalLockFileAssemblies>true</CopyLocalLockFileAssemblies> to all testing projects

From https://github.com/Particular/DotNetCoreLaunch/issues/33

Comment (dbelcham) [reply src/NServiceBus.Autofac.AcceptanceTests/NServiceBus.Autofac.AcceptanceTests.csproj]: This puts all of the files into the bin folder and allows `dotnet test` to have what it needs to run the tests. `dotnet test` is how tests are run for .NET Core on Windows and Linux in the TC build process

File: src/NServiceBus.Autofac.AcceptanceTests/NServiceBus.Autofac.AcceptanceTests.csproj [hunk 4]
```diff
@@ -1,18 +1,21 @@
 ﻿<?xml version="1.0" encoding="utf-8"?>
 <Project Sdk="Microsoft.NET.Sdk">
   <PropertyGroup>
-    <TargetFrameworks>net452;netcoreapp2.0</TargetFrameworks>
-    <DebugSymbols>True</DebugSymbols>
-    <NoWarn>PCR0001</NoWarn>
+    <TargetFrameworks>net462;netcoreapp2.0</TargetFrameworks>
```
Comment (WilliamBZA) [inline src/NServiceBus.Autofac.AcceptanceTests/NServiceBus.Autofac.AcceptanceTests.csproj]: http://nooooooooooooooo.com/

File: src/NServiceBus.Autofac.AcceptanceTests/NServiceBus.Autofac.AcceptanceTests.csproj (hunk 2 above)
Comment (boblangley) [reply src/NServiceBus.Autofac.AcceptanceTests/NServiceBus.Autofac.AcceptanceTests.csproj]: I didn't know it was in there. I will pull it out.

File: src/NServiceBus.Autofac.AcceptanceTests/NServiceBus.Autofac.AcceptanceTests.csproj (hunk 4 above)
Comment (boblangley) [reply src/NServiceBus.Autofac.AcceptanceTests/NServiceBus.Autofac.AcceptanceTests.csproj]: Reverting.

**Pull request merged**